subsys/mgmt/mcumgr/transport/src/smp_udp.c

```diff
@@ -231,98 +303,106 @@ static void create_thread(struct config *conf, const char *name)
 	k_thread_start(&(conf->thread));
 }
 
-SYS_INIT(smp_udp_init, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);
-
 int smp_udp_open(void)
 {
-	struct config *conf;
+	bool started = false;
 
 #ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV4
-	struct sockaddr_in addr4;
-
-	memset(&addr4, 0, sizeof(addr4));
-	addr4.sin_family = AF_INET;
-	addr4.sin_port = htons(CONFIG_MCUMGR_TRANSPORT_UDP_PORT);
-	addr4.sin_addr.s_addr = htonl(INADDR_ANY);
-
-	conf = &configs.ipv4;
-	conf->sock = create_socket((struct sockaddr *)&addr4, conf->proto);
-
-	if (conf->sock < 0) {
-		return -MGMT_ERR_EUNKNOWN;
+	if (!IS_THREAD_RUNNING(configs.ipv4.thread)) {
+		(void)k_sem_reset(&configs.ipv4.network_ready_sem);
+		create_thread(&configs.ipv4, "smp_udp4");
+		started = true;
```

Comment:
This isn't thread safe.

Reply:
It's user facing so they might do anything. If the interface is known as not thread safe should this be added to the Zephyr API documentation for this function?

Reply:
Fair point, note added.